src/prommis/uky/costing/tests/test_ree_costing.py

```diff
@@ -1180,6 +1224,35 @@ def test_costing_bounding(self, model):
             recalculate=True,
         )
 
+        dt = DiagnosticsToolbox(model)
+
+        try:
```

Comment:
This is a lot of different `asserts` in a single test, and the whole test will fail at the first issues (thus hiding the other checks until the first one is fixes). Generally I would recommend having separate tests for each "type" of assertion or concept, so that you can get the full picture in one pass (rather than having to fix one issue only to find a new one).